### ISSUE TITLE
Not possible to add a product with customizable options as configurable

### ISSUE THREAD

**Issue by kolaente** [magento/magento2#21471]
### Preconditions (*)
* Magento 2.2.6
* PHP `7.1.26-1+ubuntu18.04.1+deb.sury.org+1`

### Steps to reproduce (*)

1. Create a new simple product with customizable options
2. Create another new simple product serving as the configurable "root" product
3. Add the Product from the first step to the product in the second step as a configurable
4. The product now appears in the list of configurable options
4. Hit save

### Expected result (*)
* The product is successfully added as a configurable

### Actual result (*)
* The product is not added. After saving and reloading the page, it is not available in the list.

### Additional Information
* When I remove the customizable options from the product, it works as expected
* I also tried giving the "parent" product the same customizable options as their childs, didn't work
* I made sure all childs have price/quantity/stock settings correctly set

**Comment by magento-engcom-team** [comment]
Hi @kolaente. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me 2.3-develop instance` - upcoming 2.3.x release

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@kolaente do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by kolaente** [comment]
@magento-engcom-team give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @kolaente. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @kolaente, here is your [Magento instance](https://i-21471-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-21471-2-3-develop.instances.magento-community.engineering/admin](https://i-21471-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [ ] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [ ] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

-  _Next steps are available in case you are a member of **Community Maintainers**._

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by kolaente** [comment]
@magento-engcom-team I confirm the issue is reproducable on a vanilla magento instance.

**Comment by ghost** [comment]
Hi @kolaente thank you for you report, this is expected behavior, how you then can choose this options on frontend ?.

**Comment by kolaente** [comment]
@engcom-backlog-nazar How is it expected behavior if configurable options disappear without any notice? If it is really expected, it is just bad UX.
Even if, it does not seem logic to me what would prevent options from being configurable if they have customizable options.

The options are chosen in the frontend via dropdown.

**Comment by ghost** [comment]
@kolaente so please open new issue and describe problem correctly, that error message not shown, because on this issue you reported that you not able to add product with customizable options as child to configurable, that is impossible.!

**Comment by ghost** [comment]
@kolaente how you can select customizable options on configurable product with dropdown ? you chose the child in dropdown then what ? how you select customizable option ?

**Comment by kolaente** [comment]
@engcom-backlog-nazar I meant the configurable, sorry. You select the configurable product with a dropdown. I would then expect it to show the inputs for customizable options the same way with normal products. Instead, Magento just doesn't do what I expect without telling me why.

**Comment by kolaente** [comment]
A bit more context: We have a wizard where customers can customize their product very individually (they can select custom colors, upload images and a bunch of other stuff). This wizard is loaded from an iframe. When they click on "finish" inside that iframe, the software makes a request to the Magento api to put the item with all custom options in the cart. In this step, it fills all custom options the product has. 
What we wanted to do was to create a new configurable where customers could select the "base" version of the product they want to customize, then we discovered the bug.

**Comment by ghost** [comment]
@kolaente wow.. thanks for explanation, seem like you need to report this feature here -> https://github.com/magento/community-features.as  as far I know, adding a product with customizable  options to child configurable is not intentionally implemented.
